### ISSUE TITLE
Unexpected error in param jsdoc tags for methods containing arguments - 4.6 RC regression

### ISSUE THREAD

**Issue by dragomirtitian** [microsoft/TypeScript#47949]
# Bug Report

### 🔎 Search Terms

jsdoc 

### 🕗 Version & Regression Information

- This changed between versions 4.6.0-beta and 4.6.1-rc

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ts=4.7.0-dev.20220215#code/MYGwhgzhAEBCCuBzaBvAsAKGtT3sHoAqQ3PaQ6AAQAcwAnMAW1QgHtGBTAWkdYBN4IDgDoAShwCO8DhAAuAX2h1J0uaTyF866BA4A7PuKkzZACmXG5ALmjw9Aaz2sA7noCUNgG6sAln1TaZPSI8Jx6shAA3ND4+NDivJ4c0LIAFj4QADTQHHR0rHTQiKwy0GDOYACe2vKYmPJAA)

### 💻 Code

```ts
class Bug {
  
    /**
     * @param {some-module.Request} request
     */
    sendRequest(request: unknown): void {
        arguments; // Remove this, error goes away
    }

}
```

### 🙁 Actual behavior

An error occurs in the type of the `@param` jsdoc tag: `Cannot find name 'some'.`

### 🙂 Expected behavior

No error occurs on the `@param` jsdoc tag

## Notes

This only happens if `arguments` is referenced in the body of the method.

This behavior was introduced in https://github.com/microsoft/TypeScript/pull/47257 (ping @a-tarasyuk )

The issue seems to be that if `arguments` is referenced then  `getTypeFromTypeNode(lastJSDocParam.typeExpression.type)` ([Line](https://github.com/microsoft/TypeScript/pull/47257/files#diff-d9ab6589e714c71e657f601cf30ff51dfc607fc98419bf72e04f6b0fa92cc4b8R38477)) is invoked to see if the the last parameter is an array type. The problem is `getTypeFromTypeNode` will add the diagnostics about being unable to find `some` even though in this context we probably don't want to add it.
